Sources/Profile-Modify.php

```diff
@@ -2342,110 +2347,97 @@ function alert_count($memID, $unread = false)
 			'id_member' => $memID,
 		)
 	);
-
 	// First we dump alerts and possible boards information out.
-	$alerts = array();
-	$boards = array();
-	$topics = array();
-	$msgs = array();
-	$possible_boards = array();
-	$possible_topics = array();
-	$possible_msgs = array();
 	while ($row = $smcFunc['db_fetch_assoc']($request))
 	{
-		$alerts[$row['id_alert']] = !empty($row['extra']) ? $smcFunc['json_decode']($row['extra'], true) : array();
+		$alerts[$row['id_alert']] = $row;
 
+		// For these types, we need to check whether they can actually see the content.
 		if ($row['content_type'] == 'msg')
 		{
-			$possible_msgs[] = $row['content_id'];
-			$alerts[$row['id_alert']]['msg'] = $row['content_id'];
+			$alerts[$row['id_alert']]['visible'] = false;
+			$possible_msgs[$row['id_alert']] = $row['content_id'];
+		}
+		elseif (in_array($row['content_type'], array('topic', 'board')))
+		{
+			$alerts[$row['id_alert']]['visible'] = false;
+			$possible_topics[$row['id_alert']] = $row['content_id'];
 		}
-		elseif (isset($alerts[$row['id_alert']]['topic']))
-			$possible_topics[] = $alerts[$row['id_alert']]['topic'];
-		elseif (isset($alerts[$row['id_alert']]['board']))
-			$possible_boards[] = $alerts[$row['id_alert']]['board'];
+		// For the rest, they can always see it.
+		else
+			$alerts[$row['id_alert']]['visible'] = true;
 	}
 	$smcFunc['db_free_result']($request);
 
-	$possible_boards = array_unique($possible_boards);
-
-	// If this isn't the current user, get their boards.
-	if (!isset($user_info['id']) || $user_info['id'] != $memID)
+	// If we need to check board access, use the correct board access filter for the member in question.
+	if ((!isset($user_info['id']) || $user_info['id'] != $memID) && (!empty($possible_msgs) || !empty($possible_topics)))
```

Comment:
Seems like this still needs more tweaking as I am still getting the errors
"**Invalid value inserted or no type specified.(Line 2426)**"

Reply:
hmm, I noticed this also happens in the main branch, so your changes did not introduce it

Reply:
That's good to know. I'll stop banging my head against this wall and look elsewhere for the cause. Can I get you to file a new bug report about it?